conans/client/generators/cmake_find_package.py

```diff
@@ -23,6 +28,12 @@ class CMakeFindPackageGenerator(Generator):
                                           {name}_VERSION VERSION_VAR {name}_VERSION)
         mark_as_advanced({name}_FOUND {name}_VERSION)
 
+        if({name}_FIND_COMPONENTS)
```

Comment:
I am not sure about introducing this check here as it can break existing consumers using `find_package(Boost COMPONENTS whatever)` even though the COMPONENTS call does nothing (they can be doing that to avoid modifying their CMakelists?)

Reply:
I don't understand this check, a foreach that fails on the first component only?

Reply:
better remove this check.